test/integration/test-amp-story.js

```diff
@@ -0,0 +1,231 @@
+      screenX: x,
+      screenY: y,
+      clientX: x,
+      clientY: y,
+    });
+
+    win.document.elementFromPoint(x, y).dispatchEvent(event);
+  }
+
+  describesAmpStoryIntegration('with manual advancement', [
+    ampStoryPageHtml({id: 'page-0'}),
+    ampStoryPageHtml({id: 'page-1'}),
+    ampStoryPageHtml({id: 'page-2'}),
+  ], () => {
+    it('should advance in the correct order', () =>
+      expectActivePage('page-0')
+          // Forwards.
+          .then(() => expectActivePageAfterClickForward('page-1'))
+          .then(() => expectActivePageAfterClickForward('page-2'))
+
+          // Backwards.
+          .then(() => expectActivePageAfterClickBackward('page-1'))
+          .then(() => expectActivePageAfterClickBackward('page-0')))
+        .timeout(DEFAULT_TIMEOUT);
+
+    it('should activate bookend and replay on click', () =>
+      expectActivePage('page-0')
+          .then(() => expectActivePageAfterClickForward('page-1'))
+          .then(() => expectActivePageAfterClickForward('page-2'))
+          .then(() => {
+            clickForward();
+            return expectActiveBookend();
+          })
+          .then(() => {
+            win.document.querySelector(REPLAY_BUTTON_SELECTOR).click();
+            return expectActivePage('page-0');
+          }))
+        .timeout(DEFAULT_TIMEOUT);
+  });
+
+  const TIME_PAGE_0 = 800;
+  const TIME_PAGE_1 = 1500;
+  const TIME_PAGE_2 = 1000;
+
+  const TIME_ADVANCE_TIMEOUT =
+      DEFAULT_TIMEOUT +
+      TIME_PAGE_0 +
+      TIME_PAGE_1 +
+      TIME_PAGE_2 +
+      AUTO_ADVANCE_TOLERANCE_MS * 3;
+
+  describesAmpStoryIntegration('with time-based advancement', [
+    ampStoryPageHtml({id: 'page-0', 'auto-advance-after': `${TIME_PAGE_0}ms`}),
+    ampStoryPageHtml({id: 'page-1', 'auto-advance-after': `${TIME_PAGE_1}ms`}),
+    ampStoryPageHtml({id: 'page-2', 'auto-advance-after': `${TIME_PAGE_2}ms`}),
+  ], () => {
+    it('should go forwards automatically', () =>
+      expectActivePage('page-0')
+          .then(expectActivePageToLoad)
+          .then(() => expectActivePageAfterWait('page-1', TIME_PAGE_0))
+          .then(() => expectActivePageAfterWait('page-2', TIME_PAGE_1))
+          .then(() => wait(TIME_PAGE_2))
+          .then(() => expectActiveBookend(AUTO_ADVANCE_TOLERANCE_MS)))
+        .timeout(TIME_ADVANCE_TIMEOUT);
+
+    it('should allow for combined advancement', () =>
+      expectActivePage('page-0')
+          .then(expectActivePageToLoad)
+          .then(() => expectActivePageAfterWait('page-1', TIME_PAGE_0))
+          .then(() => expectActivePageAfterClickBackward('page-0'))
+          .then(() => expectActivePageAfterWait('page-1', TIME_PAGE_0))
+          .then(() => expectActivePageAfterClickForward('page-2'))
+          .then(() => {
+            clickForward();
+            return expectActiveBookend();
+          }))
+        .timeout(TIME_ADVANCE_TIMEOUT);
+  });
+
+  // TODO(alanorozco): Test for media-based advancement
```

Comment:
Heads up that SauceLabs uses VM's that can't play most videos that require hardware acceleration. (https://support.saucelabs.com/hc/en-us/articles/115002202568-Unsupported-Video-Types). Might be a good time to consider using real devices ( https://github.com/ampproject/amphtml/issues/7822 ) @rsimha-amp

Reply:
Noted. Will try to look at this soon.